test/common/index.js

```diff
@@ -75,9 +76,35 @@ exports.mustNotCall = function (msg) {
   };
 };
 
-exports.runTest = async function (test, buildType, buildPathRoot = process.env.BUILD_PATH || '') {
-  buildType = buildType || process.config.target_defaults.default_configuration || 'Release';
+const buildTypes = {
+  Release: 'Release',
+  Debug: 'Debug'
+};
+
+async function checkBuildType (buildType) {
+  try {
+    await access(path.join(path.resolve('./test/build'), buildType));
+    return true;
+  } catch {
+    return false;
+  }
+}
 
+async function whichBuildType () {
```

Comment:
What about determining with a simple environ like `BUILD_TYPE`?

Reply:
Note that if people incremental build with `node-gyp build -C test --debug`, the build directory will not be cleaned and `build/Release` can still present.